pkgs/build-support/fake-uname/default.nix

```diff
@@ -0,0 +1,34 @@
+# expr and script based on our lsb_release
+{ stdenv
+, lib
+, substituteAll
+, coreutils
+, getopt
+, modDirVersion ? ""
+}:
+
+substituteAll {
+  name = "uname";
+
+  src = ./fake-uname.sh;
+
+  dir = "bin";
+  isExecutable = true;
+
+  inherit coreutils getopt;
+
+  uSystem = stdenv.buildPlatform.uname.system;
```

Comment:
Does this eval correctly if kernel is none? I never know exactly how that works.

Reply:
fix'd